pysat/instruments/jro_isr.py

```diff
@@ -201,12 +201,13 @@ def clean(self):
                 print('WARNING: this level 2 data has no quality flags')
 
             ida, = np.where((self.data.indexes['gdalt'] > 200.0))
+            idx['gdalt'] = np.unique(ida)
         else:
             print("WARNING: interpretation of drifts below 200 km should " +
                   "always be done in partnership with the contact people")
             
     # downselect data based upon cleaning conditions above
-    self.data = self.data.isel(gdalt=ida)
+    self.data = self[idx]
```

Comment:
Thanks for looking into this. I'm going to try and poke around in this area myself, see if pysat can help out more.

Reply:
The thing to do might be to allow a dictionary of {"time": [0, 1, 2, ..., n]} to work for pandas data in addition to just the list of indices.  Then we could default to a dictionary specification everywhere.